otx/core/data/adapter/base_dataset_adapter.py

```diff
@@ -162,13 +163,15 @@ def get_label_schema(self) -> LabelSchemaEntity:
     def _get_subset_data(self, subset: str, dataset: DatumDataset) -> DatumDatasetSubset:
         """Get subset dataset according to subset."""
         with eager_mode(True, dataset):
-            for k, v in dataset.subsets().items():
-                if subset in k or "default" in k:
-                    v = v.as_dataset()
-                    return v
-                if subset == "test" and "val" in k:
-                    v = v.as_dataset()
-                    return v
+            subsets = list(dataset.subsets().keys())
+
+            for s in [subset, "default"]:
+                if subset == "test":
+                    s = "val"
+                exact_subset = get_close_matches(s, subsets)
```

Comment:
Does it find a closest key name in dataset_subsets with subset from argument?

Reply:
Yes it does